hs/src/Reach/Verify/SMT.hs

```diff
@@ -1344,6 +1349,20 @@ smt_n = \case
         smt_invblock (B_Assume True) inv
         smt_invblock (B_Assume False) cond
         smt_n k
+
+      insertConstrainedMaps :: S.Set DLMVar -> S.Set DLMVar
+      insertConstrainedMaps = insertConstrainedMaps' invTail
+      DLBlock _ _ invTail _ = inv
+
+      insertConstrainedMaps' :: DLTail -> S.Set DLMVar -> S.Set DLMVar
+      insertConstrainedMaps' dltail maps = case dltail of
+        DT_Return _ -> maps
+        DT_Com stmt dltail' ->
+          let tailMaps = insertConstrainedMaps' dltail' maps in
+            case stmt of
+              DL_MapReduce _ _ _ dlmv _ _ _ _ -> S.insert dlmv tailMaps
```

Comment:
I think it would be better to stick something in the ctxt and then in the code that runs on `DL_MapReduce`, record that it was looked at it so there's no another walking of the code

Reply:
gotcha